src/FFmpegReader.cpp

```diff
@@ -706,9 +706,16 @@ void FFmpegReader::UpdateVideoInfo() {
 	info.vcodec = pCodecCtx->codec->name;
 	info.video_bit_rate = (pFormatCtx->bit_rate / 8);
 
-	// set frames per second (fps)
-	info.fps.num = pStream->avg_frame_rate.num;
-	info.fps.den = pStream->avg_frame_rate.den;
+	// Frame rate from the container and codec
+	AVRational framerate = av_guess_frame_rate(pFormatCtx, pStream, NULL);
+	info.fps.num = framerate.num;
+	info.fps.den = framerate.den;
```

Comment:
This reminds me that I tried to add operators and conversion methods into `openshot::Fraction` so it would automatically convert from/to `AVRational` as needed, once. But I think the fact that `AVRational` is implemented in C instead of C++ probably doomed that idea right from the start.